File: 933.cpp
#include<iostream>
#include<queue>

using namespace std;

class RecentCounter {
public:
    RecentCounter(): v(), curr(0){}
    int ping(int t){
        v.push_back(t);
        while(v[curr] < t - 3000) curr++;
        return v.size() - curr;
    }
private:
    vector<int> v;
    int curr;
};

File: python/933.py
class RecentCounter:
    v = []
    curr = 0
    def __init__(self):
        self.v = []
        self.curr = 0
        
    def ping(self, t: int) -> int:
        self.v.append(t)
        while self.v[self.curr] < t - 3000:
            self.curr += 1
        return len(self.v) - self.curr